firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -42,40 +89,68 @@ void moveSpaces(int motor[], int spaces, int dir)
   0         1        2        3
   Step pin, Dir pin, MS1 pin, MS2 pin
 */
-void setScale(int motor[], int scale) 
+
+void moveStraight(int motor[], int startCol, int startRow, int endCol, int endRow)
 {
-  if (scale == 1)  // whole steps
-  {
-    digitalWrite(motor[2], LOW);
-    digitalWrite(motor[3], LOW);
-  }
-  else if (scale == 2)  // 1/2 steps
+  // A specific motor is passed to this function since we are only moving one here
+  
+  // How many steps per space
+  int spaces, dir, numSteps;
+  int i;
+
+  // This could be two cases, x or y movement
+  if (endRow == startRow)
   {
-    digitalWrite(motor[2], HIGH);
-    digitalWrite(motor[3], LOW);
+    // X movement
+    spaces = abs(endCol - startCol);
+    dir = (endCol > startCol) ? RIGHT : LEFT;
+    setScale(xMotor, 1);
   }
-  else if (scale == 4)  // 1/4 steps
+  else if (endCol == startCol)
   {
-    digitalWrite(motor[2], LOW);
-    digitalWrite(motor[3], HIGH);
+    // Y movement
+    spaces = abs(endRow - startRow);
+    dir = (endRow > startRow) ? UP : DOWN;
+    setScale(yMotor, 1);
   }
-  else if (scale == 8)  // 1/8 steps
+
+  numSteps = spaces * stepsPerSpace;
+
+  // Enable motor driver inputs/output
+  enableMotors();
+
+  // Set direction of motor
+  digitalWrite(motor[1], dir);
+
+  // Rotate motor some number of steps
+  for (i = 0; i < numSteps; i++) 
   {
-    digitalWrite(motor[2], HIGH);
-    digitalWrite(motor[3], HIGH);
+    if (digitalRead(X_AXIS_ENDSTOP_SWITCH) == HIGH  ||  digitalRead(Y_AXIS_ENDSTOP_SWITCH))
+      break;
```

Comment:
When breaking because we hit the endstop, should we do some error handling? I imagine we shouldn't hit the end stops when calling a move from location A to location B. Even if you don't handle this now, maybe you can add a TODO?

Reply:
This is being addressed in #73